lib/fastlane/setup/setup_ios.rb

```diff
@@ -52,6 +45,41 @@ def run
       # rubocop:enable Lint/RescueException
     end
 
+    def default_setup(path: nil)
+      copy_existing_files
+      generate_appfile(manually: false)
+      detect_installed_tools # after copying the existing files
+      create_app_if_necessary
+      enable_deliver
+      FileUtils.mkdir(path)
+      generate_fastfile(manually: false)
+      show_analytics
+    end
+
+    def manual_setup(path: nil)
+      copy_existing_files
+      generate_appfile(manually: true)
+      detect_installed_tools # after copying the existing files
+      ask_to_enable_other_tools
+      FileUtils.mkdir(path)
+      generate_fastfile(manually: true)
+      show_analytics
+    end
+
+    def ask_to_enable_other_tools
+      unless self.itc_ref && self.portal_ref
+        wants_to_create_app = agree('Would you like to create your app on iTunes Connect and the Developer Portal?', true)
+        if wants_to_create_app
+          create_app_if_necessary
+          detect_if_app_is_available # check if the app was, in fact, created.
+        end
+      end
+      if self.itc_ref && self.portal_ref
```

Comment:
thats why i added another call to `detect_if_app_is_available` after `create_app_if_necessary`